packages/ai/core/util/chat-store.test.ts

```diff
@@ -0,0 +1,1090 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import type { UIMessage } from '../types';
+import { type ChatState, ChatStore } from './chat-store';
+
+const initStore = (chats?: Record<string, Pick<ChatState, 'messages'>>) => {
+  const store = new ChatStore({ chats });
+  return store;
+};
```

Comment:
is this needed? might make sense to just call `new ChatStore` in the tests to be explicit.

Reply:
(then you can also inline some variables in the tests to see how the store would look like)